conda-store-server/conda_store_server/app.py

```diff
@@ -107,20 +108,17 @@ class CondaStore(LoggingConfigurable):
     )
 
     build_key_version = Integer(
-        2,
+        BuildKey.set_current_version(2),
         help="Build key version to use: 1 (long, legacy), 2 (short, default)",
         config=True,
     )
 
     @validate("build_key_version")
     def _check_build_key_version(self, proposal):
-        expected = [1, 2]
-        if proposal.value not in expected:
-            raise TraitError(
-                f"c.CondaStore.build_key_version: invalid build key version: "
-                f"{proposal.value}, expected: {expected}"
-            )
-        return proposal.value
+        try:
+            return BuildKey.set_current_version(proposal.value)
+        except Exception as e:
+            raise TraitError(f"c.CondaStore.build_key_version: {e}")
```

Comment:
The current version is now set right after `build_key_version` is processed, which is better than before, when it was done only when creating a DB session.

Reply:
Is it better to use `raise X from e` here?

Reply:
In general, yes. But it doesn't work with traitlets. E.g.,:

```py
            raise TraitError("c.CondaStore.build_key_version") from e
```

would print

```
[CondaStoreServer] CRITICAL | Bad config encountered during initialization: c.CondaStore.build_key_version
```

No traceback is included. No additional information is printed.